packages/vite/src/node/plugins/define.ts

```diff
@@ -122,23 +119,34 @@ export function definePlugin(config: ResolvedConfig): Plugin {
         return
       }
 
-      const [define, pattern, banner] = ssr ? ssrPattern : defaultPattern
+      const [define, pattern, importMetaEnvVal] = ssr
+        ? ssrPattern
+        : defaultPattern
       if (!pattern) return
 
       // Check if our code needs any replacements before running esbuild
       pattern.lastIndex = 0
       if (!pattern.test(code)) return
 
+      let marker = importMetaEnvMarker
+      while (new RegExp(escapeRegex(marker)).test(code)) {
+        marker += '$'
+      }
```

Comment:
Could we do `code.includes(marker)` instead? That should make this part more performant.